libkbfs/block_journal.go

```diff
@@ -47,20 +49,16 @@ import (
 // ID, and key_server_half, which contains the raw data for the
 // associated key server half.
 //
-// TODO: Add a mode which doesn't assume that this is the only storage
-// for TLF data, i.e. that doesn't remove files when the (known)
-// refcount drops to zero, etc.
-type bserverTlfJournal struct {
+// blockJournal is not goroutine-safe, so any code that uses it must
```

Comment:
I don't totally understand the point of moving the lock into the upper layer.  It seems like you want to do that so there's only lock in the bundle that covers both the MD and bserver journals, right?  Is that solving a real race?  And even that's important, why not leave a lock here as well, so we don't have to rely on `bserver_disk` to have a lock?